test/object.cc

```diff
@@ -39,14 +39,26 @@ void DefineProperties(const CallbackInfo& info) {
       PropertyDescriptor::Function("function", TestFunction),
     });
   } else if (nameType.Utf8Value() == "string") {
+    // VS2013 has lifetime issues when passing temporary objects into the constructor of another
+    // object. It generates code to destruct the object as soon as the constructor call returns.
+    // Since this isn't a common case for using std::string objects, I'm refactoring the test to
+    // work around the issue.
+    std::string str1("readonlyAccessor");
+    std::string str2("readwriteAccessor");
+    std::string str3("readonlyValue");
+    std::string str4("readwriteValue");
+    std::string str5("enumerableValue");
+    std::string str6("configurableValue");
+    std::string str7("function");
+
     obj.DefineProperties({
-      PropertyDescriptor::Accessor(std::string("readonlyAccessor"), TestGetter),
-      PropertyDescriptor::Accessor(std::string("readwriteAccessor"), TestGetter, TestSetter),
-      PropertyDescriptor::Value(std::string("readonlyValue"), trueValue),
-      PropertyDescriptor::Value(std::string("readwriteValue"), trueValue, napi_writable),
-      PropertyDescriptor::Value(std::string("enumerableValue"), trueValue, napi_enumerable),
-      PropertyDescriptor::Value(std::string("configurableValue"), trueValue, napi_configurable),
-      PropertyDescriptor::Function(std::string("function"), TestFunction),
+      PropertyDescriptor::Accessor(str1, TestGetter),
```

Comment:
@jasongin there is already an overload for const char *, that's what the other leg of the conditional tests. It just gets chained to for the std::string constructor. Does that resolve your concerns?